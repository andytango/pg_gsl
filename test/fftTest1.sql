/* 
 * Copyright (C) 2016 - Ferndale-Hall (pg_gsl@ferndale-hall.co.uk)
 *
 * This file is part of pg_gsl.
 *
 * pg_gsl is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * pg_gsl is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with pg_gsl.  If not, see <http://www.gnu.org/licenses/>.
 *
 */

 /* Call the pg_gsl_x_version() function */

select pg_gsl_x_version();

/*
 * Call the transform with a square wave, the same as the fft example
 * in the GSL documentation at:
 * https://www.gnu.org/software/gsl/manual/html_node/Mixed_002dradix-FFT-routines-for-real-data.html
 */
select pg_gsl_fft_real_transform('{
	0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,
	0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,
	0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,
	0.0, 0.0, 0.0,
	1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0,
	1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0,
	1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0,
	1.0, 1.0, 1.0, 0.0,
	0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,
	0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,
	0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,
	0.0, 0.0, 0.0}'::double  precision[])::decimal(7, 5)[];

/*
 * Call the transform awith a square wave, and perform the reverse transform t
 * reconstruct the square wave, the same as the fft example
 * in the GSL documentation at:
 * https://www.gnu.org/software/gsl/manual/html_node/Mixed_002dradix-FFT-routines-for-real-data.html
 */

select pg_gsl_fft_halfcomplex_inverse(pg_gsl_fft_real_transform('{
	0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,
	0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,
	0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,
	0.0, 0.0, 0.0,
	1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0,
	1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0,
	1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0,
	1.0, 1.0, 1.0, 0.0,
	0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,
	0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,
	0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,
	0.0, 0.0, 0.0}'::double  precision[]))::decimal(7, 5)[];
