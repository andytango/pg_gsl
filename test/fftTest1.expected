# 
# Copyright (C) 2016 - Ferndale-Hall (pg_gsl@ferndale-hall.co.uk)
#
# This file is part of pg_gsl.
#
# pg_gsl is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
#
# pg_gsl is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with pg_gsl.  If not, see <http://www.gnu.org/licenses/>.
#


 pg_x_gsl_version 
------------------
 0.0.1-dirty
(1 row)

                                                                                                                                                                                                                                                                                                                                                                                                                                   pg_gsl_fft_real_transform                                                                                                                                                                                                                                                                                                                                                                                                                                    
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 {33.00000,-27.34870,-1.72063,13.84599,1.74915,-0.32786,-0.06254,-6.52502,-1.67534,5.41695,1.76007,-0.31156,-0.12336,-3.43061,-1.61432,3.18833,1.75280,-0.28481,-0.18074,-2.11803,-1.53884,2.08758,1.72700,-0.24819,-0.23306,-1.36196,-1.45034,1.39195,1.68257,-0.20254,-0.27877,-0.85700,-1.35041,0.89044,1.61971,-0.14890,-0.31642,-0.49127,-1.24080,0.50000,1.53884,-0.08851,-0.34473,-0.21429,-1.12336,0.18200,1.44065,-0.02281,-0.36257,0.00000,-1.00000,-0.08343,1.32608,0.04662,-0.36900,0.16648,-0.87269,-0.30716,1.19629,0.11803,-0.36327,0.29434,-0.74342,-0.49535,1.05267,0.18959,-0.34486,0.38976,-0.61416,-0.65156,0.89680,0.25933,-0.31348,0.45716,-0.48683,-0.77784,0.73044,0.32524,-0.26906,0.50000,-0.36327,-0.87534,0.55551,0.38522,-0.21177,0.52114,-0.24523,-0.94467,0.37402,0.43715,-0.14204,0.52312,-0.13431,-0.98618,0.18812,0.47890,-0.06050,0.50829,-0.03198,-1.00000}
(1 row)

                                                                                                                                                                                                                                                                                                                                                                                                  pg_gsl_fft_halfcomplex_inverse                                                                                                                                                                                                                                                                                                                                                                                                   
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 {0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,1.00000,1.00000,1.00000,1.00000,1.00000,1.00000,1.00000,1.00000,1.00000,1.00000,1.00000,1.00000,1.00000,1.00000,1.00000,1.00000,1.00000,1.00000,1.00000,1.00000,1.00000,1.00000,1.00000,1.00000,1.00000,1.00000,1.00000,1.00000,1.00000,1.00000,1.00000,1.00000,1.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000,0.00000}
(1 row)

